virtualmachine/vsphere/vm.go

```diff
@@ -735,7 +741,16 @@ func getNicInfo(vmMo mo.VirtualMachine) []VirtualEthernetCard {
 			nwcard := c.GetVirtualEthernetCard()
 			nic.MacAddress = nwcard.MacAddress
 			nic.NicName = nwcard.DeviceInfo.GetDescription().Label
-			nic.NetworkName = nwcard.Backing.(*types.VirtualEthernetCardNetworkBackingInfo).DeviceName
+			switch b := nwcard.Backing.(type) {
+			case *types.VirtualEthernetCardNetworkBackingInfo:
+				nic.NetworkName = b.DeviceName
+			case *types.VirtualEthernetCardDistributedVirtualPortBackingInfo:
+				nic.SwitchUuid = b.Port.SwitchUuid
```

Comment:
Although this looks harmless -- but if it is not going to be used as it is -- we could get rid of this code.